### PR TITLE
Including GeneOCR API

### PR DESCRIPTION
**Pull request by paulgeser**
Add geneocr api

### CODE REVIEW

File: geneocr-api/1.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#    libfile-slurp-perl && \
+#    apt-get autoclean && rm -rf /var/lib/apt/lists/*
+
+RUN git clone --depth=1 https://github.com/bedapub/geneocr-ui.git && \
+    cd geneocr-ui && \
+    ls
+
+WORKDIR /code
+
+COPY ./app/requirements.txt /code/requirements.txt
```
Comment (osallou) [inline geneocr-api/1.0/Dockerfile]: COPY does not work here, this is not a local file, but I suppose it is in what you clone , I suppose you expect a RUN cp ....

File: geneocr-api/1.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+RUN apt-get update
+RUN apt-get install ffmpeg libsm6 libxext6 libleptonica-dev tesseract-ocr libtesseract-dev python3-pil tesseract-ocr-eng tesseract-ocr-script-latn  -y
+
+RUN pip install --upgrade pip
+RUN pip install --no-cache-dir --upgrade -r /code/requirements.txt
+
+COPY ./app /code/app
```
Comment (osallou) [inline geneocr-api/1.0/Dockerfile]: same comment as above

**Review by osallou — CHANGES_REQUESTED** [review]
you should first try to build to docker locally...

File: geneocr-api/1.0/Dockerfile [hunk 3]
```diff
@@ -39,24 +39,3 @@ WORKDIR /geneocr-api/app
 EXPOSE 8000
 
 CMD ["uvicorn", "app:app", "--host", "0.0.0.0", "--port", "8000"]
```
Comment (osallou) [inline geneocr-api/1.0/Dockerfile:43]: not missing the pip install requirements.txt ?

Comment (paulgeser) [reply geneocr-api/1.0/Dockerfile:43]: They are being installed on line 30

Comment (osallou) [reply geneocr-api/1.0/Dockerfile:43]: and build fails

**Review by osallou — APPROVED** [review]

**Comment by osallou** [comment]
build fails at: : The command '/bin/sh -c git clone --depth=s1 https://github.com/bedapub/geneocr-api.git &&     cd geneocr-api &&     ls' returned a non-zero code: 128
I suppose you mean --depth=s1  => =1

**Comment by osallou** [comment]
@paulgeser merged :-)

**Comment by osallou** [comment]
As a remark, this container is very large, more than 2gb

**Comment by paulgeser** [comment]
> 



> As a remark, this container is very large, more than 2gb

Thanks for the remark, will take a look in the future to reduce it!

**Comment by paulgeser** [comment]
Thanks for all the assistance @osallou

**Pull request merged**